apps/04TowersOfHanoi.js

```diff
@@ -12,31 +12,52 @@ var stacks = {
   b: [],
   c: []
 };
+// var popped = stacks[startStack].pop();
 
 function printStacks() {
   console.log("a: " + stacks.a);
   console.log("b: " + stacks.b);
   console.log("c: " + stacks.c);
 }
 
-function movePiece() {
-  // Your code here
-
+function movePiece(startStack, endStack) {
+//VARIABLE TO HOLD POPPED ITEM
+  var popped = stacks[startStack].pop();
+//ADD THAT ITEM TO USER SELECTED STACK
+  stacks[endStack].push(popped);
 }
 
-function isLegal() {
-  // Your code here
-
+function isLegal(startStack, endStack) {
+//CREATE NEW VARIABLES TO REDUCE AMOUNT OF CODE TO TYPE IN THE LONG RUN
+  var tower1 = stacks[startStack];
+  var tower2 = stacks[endStack];
+//ALLOW ANY PIECE TO BE PLACED IN AN EMPTY STACK
+  if (tower2 == '') {
+    return true;
+  }
+//CHECK TO SEE IF START PIECE IS SMALLER THAN END PIECE THAT IT IS BEING PLACED ON
+  if (tower1[tower1.length - 1] < tower2[tower2.length-1]) {
+    return true;
+  } else {
+      console.log('Ahh Ahh Ah, you didnt say the magic word!!')
```

Comment:
I forget what my previous comment was but make sure you are indenting everything from the left margin not the word of the else
so in this case you have indented 2 spaces from the "e" but you should be indenting base on the left margin of the entire file. 
so line 40 and line 42 should be indented the same